### DISCUSSION TITLE
How to shrink/grow other element while dragging bottom sheet

### DISCUSSION THREAD

**Discussion by Arilith** [gorhom/react-native-bottom-sheet#1082] [Q&A]
Hi,

Currently I've got a map visible behind (above) my bottom sheet, and what I'd like is that it dynamically scales when dragging the bottom sheet up (or down)

![image](https://user-images.githubusercontent.com/13538135/187659926-b11d476e-1694-43d3-ba2d-9209f9570255.png)

Currently I've just set it so that when the sheet opens to a new snap point, the height of the map is changed. However, this doesn't look that smooth as it dynamically resizing on drag. Is there any way to implement this?

I've found `useGestureEventsHandlersDefault`, and `gestureEventsHandlersHook`, but I would have no clue how to implement that. I'm quite new to react native, so this could be very easy 😅 

Thanks in advance!

**Comment by AfzalAghariya** [comment]
@Arilith Did you found any solution? 
I am looking for this same behaviour in my current App

**Reply by danieljvdm** [reply]
@AfzalAghariya it's pretty easy, just use `animatedIndex` or `animatedPosition` props in `BottomSheet` and interpolate those for an animated value that scales your map

**Reply by AfzalAghariya** [reply]
Thanks @danieljvdm I'll try this approach.

**Reply by arturschwarz** [reply]
@danieljvdm 
Can you post a code example, please? 🙏🏻 

@AfzalAghariya Were you successful trying this approach?

**Reply by chikamichi** [reply]
See https://github.com/gorhom/react-native-bottom-sheet/discussions/2128 for a simple solution using reanimated’s `useAnimatedReaction`.

It may be "simplified" to merely tracking `animatedIndex` or `animatedPosition`, as is examplified in [Custom Handle](https://gorhom.dev/react-native-bottom-sheet/custom-handle), yet I feel like `useAnimatedReaction` is more straightforward and powerful.
